ERCS/erc-7595.md

```diff
@@ -0,0 +1,491 @@
+
+	/**
+	 * @dev Amount of tokens that was already taken from the disperse.
+	 *
+	 * @param address address of token.
+	 * @return uint256 total amount of tokens already taken.
+	 */
+	function disperseTotalTaken(address tokenAddress) external view returns (uint256);
+
+	/**
+	 * @dev Amount of disperse already taken by each tokenId.
+	 *
+	 * @param tokenId unique identifier of unit.
+	 * @param address address of token.
+	 * @return uint256 amount of tokens already taken.
+	 */
+	function disperseTaken(uint256 tokenId, address tokenAddress) external view returns (uint256);
+
+	/**
+	 * @dev Mapping of `tokenId`s to token addresses that have collateralized before.
+	 *
+	 * @param tokenId unique identifier of unit.
+	 * @param index in array.
+	 * @return address address of token.
+	 */
+	function collateralTokens(uint256 tokenId, uint256 index) external view returns (address);
+
+	/**
+	 * @dev Token balances that are stored under `tokenId`.
+	 *
+	 * @param tokenId unique identifier of unit.
+	 * @param address address of token.
+	 * @return uint256 token balance.
+	 */
+	function collateralBalances(uint256 tokenId, address tokenAddress) external view returns (uint256);
+
+	/**
+	 * @dev Calculator function for harvesting.
+	 *
+	 * @param amount of `communityToken`s to spend
+	 * @param address address of token to be harvested
+	 * @return amount to harvest based on inputs
+	 */
+	function getAmount(uint256 amount, address tokenAddress) external view returns (uint256);
+
+	/**
+	 * @dev Collect commission fees gathered in exchange for `communityToken`.
+	 *
+	 * @param amounts[] array of amounts to collateralize
+	 * @param address[] array of token addresses
+	 */
+	function harvest(uint256[] memory amounts, address[] memory tokenAddresses) external;
+
+	/**
+	 * @dev Collateralize NFT with different tokens and amounts.
+	 *
+	 * @param tokenId unique identifier for specific NFT
+	 * @param amounts[] array of amounts to collateralize
+	 * @param address[] array of token addresses
+	 */
+	function collateralize(
+		uint256 tokenId,
+		uint256[] memory amounts,
+		address[] memory tokenAddresses
+	) external payable;
+
+	/**
+	 * @dev Withdraw underlying collateral.
+	 *
+	 * Requirements:
+	 * - only owner of NFT
+	 *
+	 * @param tokenId unique identifier for specific NFT
+	 * @param amounts[] array of amounts to collateralize
+	 * @param address[] array of token addresses
+	 */
+	function uncollateralize(
+		uint256 tokenId, 
+		uint256[] memory amounts, 
+		address[] memory tokenAddresses
+	) external;
+
+	/**
+	 * @dev Split collateral among all existent tokens.
+	 *
+	 * @param amounts[] to be dispersed among all NFT owners
+	 * @param address[] address of token to be dispersed
+	 */
+	function disperse(uint256[] memory amounts, address[] memory tokenAddresses) external payable;
+}
+```
+
+### Abstraction layer for already deployed NFT collections
+
+```solidity
+
+interface IEnviousHouse {
```

Comment:
Is this name a placeholder, or is this the intended final name for the interface?

Reply:
IEnviousHouse is the final name of the interface